.github/workflows/test-dotnet.yml

```diff
@@ -4,18 +4,18 @@ on: [ pull_request, push ]
 
 jobs:
   setup-version:
-    runs-on: ${{ matrix.operating-system }}
+    runs-on: ubuntu-latest
     strategy:
       fail-fast: false
-      matrix:
-        operating-system: [ubuntu-latest ]
-        dotnet-version: ['5.0']
-    name: Dotnet ${{ matrix.dotnet }} sample
+    name: dotnet test
     steps:
       - uses: actions/checkout@v2
       - name: Setup dotnet
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: ${{ matrix.dotnet }}
-      - run: dotnet build RecordParser.Test/RecordParser.Test.csproj
```

Comment:
and no need to build prior to testing, the `dotnet test` command already restores and builds the project implicitly!